nixos/modules/services/backup/syncoid.nix

```diff
@@ -197,14 +197,14 @@ in {
                ])) (attrValues cfg.commands);
         after = [ "zfs.target" ];
         serviceConfig = {
-          ExecStartPre = (map (pool: lib.escapeShellArgs [
-            "+/run/booted-system/sw/bin/zfs" "allow"
-            cfg.user "hold,send" pool
-          ]) (getPools "source")) ++
-          (map (pool: lib.escapeShellArgs [
-            "+/run/booted-system/sw/bin/zfs" "allow"
-            cfg.user "create,mount,receive,rollback" pool
-          ]) (getPools "target"));
+          ExecStartPre = let
+            allowCmd = permissions: pool: lib.escapeShellArgs [
+              "+/run/booted-system/sw/bin/zfs" "allow"
+              cfg.user (concatStringsSep "," permissions) pool
+            ];
+          in
+            (map (allowCmd ([ "hold" "send" ] ++ optionals (!(elem "--no-sync-snap" cfg.commonArgs)) [ "snapshot" "destroy" ])) (getPools "source")) ++
```

Comment:
If we do this conditionally, the logic needs to be more complicated and incorporate the `extraArgs` and `useCommonArgs` options. On the other hand, perhaps we should just add these permissions unconditionally.

Reply:
Pushed a new change where I opted to just add those permissions unconditionally, for simplicity.